File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.py
def solution(nums):
    answer = 0

    n = int(len(nums)/2)
    
    l = len(set(nums))
    print(n,l)
    
    if n>l:
        answer = l
    else:
        answer = n
    
    return answer



















# def solution(nums):
#     answer = 0
#     set_nums = set(nums)
#     print(set_nums)
#     answer = len(set_nums)  
#     if answer > len(nums)/2:
#         answer = int(len(nums)/2)
#     return answer

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;
class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        
        List<Integer> numsList = new ArrayList<>();
        
        for (int n : nums) numsList.add(n);
        
        
        int N = nums.length/2;
        Set<Integer> set = new HashSet<>(numsList);
        
        if (set.size() >=N){
            return N;
        }
        else{
            return set.size();
        }
        
    }
}